packages/wrangler/src/__tests__/patch-config.test.ts

```diff
@@ -0,0 +1,880 @@
+				},
+			],
+		},
+		replacingPatch: {
+			kv_namespaces: [
+				{
+					binding: "KV",
+					id: "1234",
+				},
+			],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+
+				[[kv_namespaces]]
+				binding = "KV"
+				id = "1234"
+
+				`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"kv_namespaces": [
+						{
+							"binding": "KV",
+							"id": "1234"
+						}
+					]
+				}
+			`,
+	},
+	{
+		name: "delete an existing binding so that none are left",
+		original: {
+			kv_namespaces: [
+				{
+					binding: "KV",
+				},
+			],
+		},
+		replacingPatch: {
+			kv_namespaces: undefined,
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+
+				`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name"
+				}
+			`,
+	},
+	// This one doesn't work because the jsonc-parser leaves behind a stray bracket when deleting
+	// there are possibly solutions that I am not inclined to solve right now
+	// e.g. passing in {binding: undefined} in the patch instead and cleaning up empty objects
+	// {
+	// 	name: "delete an existing binding (but some bindings of that type are still left)",
+	// 	original: {
+	// 		kv_namespaces: [
+	// 			{
+	// 				binding: "KV",
+	// 			},
+	// 			{
+	// 				binding: "KV2",
+	// 			},
+	// 		],
+	// 	},
+	// 	replacingPatch: {
+	// 		kv_namespaces: [
+	// 			{
+	// 				binding: "KV",
+	// 			},
+	// 			undefined,
+	// 		],
+	// 	},
+	// 	expectedToml: dedent`
+	// 			compatibility_date = "2022-01-12"
+	// 			name = "test-name"
+
+	// 			[[kv_namespaces]]
+	// 			binding = "KV"
+
+	// 			`,
+	// 	expectedJson: dedent`
+	// 			{
+	// 				"compatibility_date": "2022-01-12",
+	// 				"name": "test-name",
+	// 				"kv_namespaces": [
+	// 					{
+	// 						"binding": "KV"
+	// 					}
+	// 				]
+	// 			}
+	// 		`,
+	// },
+	{
+		name: "edit a compat flag",
+		original: {},
+		replacingPatch: {
+			compatibility_flags: ["no_nodejs_compat"],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+				compatibility_flags = [ "no_nodejs_compat" ]
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"compatibility_flags": [
+						"no_nodejs_compat"
+					]
+				}
+			`,
+	},
+	{
+		name: "add a compat flag",
+		original: { compatibility_flags: ["nodejs_compat"] },
+		replacingPatch: {
+			compatibility_flags: ["nodejs_compat", "flag"],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+				compatibility_flags = [ "nodejs_compat", "flag" ]
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"compatibility_flags": [
+						"nodejs_compat",
+						"flag"
+					]
+				}
+			`,
+	},
+	{
+		name: "delete a compat flag",
```

Comment:
There don't seem to be any compat flags in the original here?

Reply:
argh excellent point, got my tests mixed up [6c2ce25](https://github.com/cloudflare/workers-sdk/pull/7521/commits/6c2ce25fd009b90bb124783d827bedb069d12fe9)